clang/tools/clang-scan-deps/Opts.td

```diff
@@ -27,6 +27,9 @@ defm compilation_database : Eq<"compilation-database", "Compilation database">;
 defm module_name : Eq<"module-name", "the module of which the dependencies are to be computed">;
 defm dependency_target : Eq<"dependency-target", "The names of dependency targets for the dependency file">;
 
+defm tu_path: Eq<"tu-path", "The path to the translation unit for depscan. Not compatible with -module-name">;
```

Comment:
I'm a bit confused by the command-line API. Why do we duplicate `-module-name` and `-import-module`? Aren't they the same? (Except that `-import-module` allows you to list more than one module name.)

I'd also prefer if `clang-scan-deps` stopped amending the compilation database. Each of its entries needs to have a TU path, which are typically distinct, so overriding them all to single value specified by `-tu-path` seems incorrect. (We already allow `-module-name` with CDBs, which feels wrong as well.)

I think that for `-tu-path` and `-import-module` (and `-module-name`), it would make sense to only allow those with the `clang-scan-deps <non-input-flags...> -- <clang-command-line-for-single-tu>` format used in the `P1689.cppm` test.